Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
be = xr.open_dataset('../../Land_and_Ocean_LatLong1.nc')
#be = xr.open_dataset('/Users/vonw/work/courses/2021-2022/spring/CE401/jupyterbook/CE401/Land_and_Ocean_LatLong1.nc')
be
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mKeyError[0m                                  Traceback (most recent call last)
File [0;32m/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/xarray/backends/file_manager.py:209[0m, in [0;36mCachingFileManager._acquire_with_cache_info[0;34m(self, needs_lock)[0m
[1;32m    208[0m [38;5;28;01mtry[39;00m:
[0;32m--> 209[0m     file [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_cache[49m[43m[[49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_key[49m[43m][49m
[1;32m    210[0m [38;5;28;01mexcept[39;00m [38;5;167;01mKeyError[39;00m:

File [0;32m/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/xarray/backends/lru_cache.py:55[0m, in [0;36mLRUCache.__getitem__[0;34m(self, key)[0m
[1;32m     54[0m [38;5;28;01mwith[39;00m [38;5;28mself[39m[38;5;241m.[39m_lock:
[0;32m---> 55[0m     value [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_cache[49m[43m[[49m[43mkey[49m[43m][49m
[1;32m     56[0m     [38;5;28mself[39m[38;5;241m.[39m_cache[38;5;241m.[39mmove_to_end(key)

[0;31mKeyError[0m: [<function _open_scipy_netcdf at 0x7f14752bc940>, ('/home/runner/work/CE401/Land_and_Ocean_LatLong1.nc',), 'r', (('mmap', None), ('version', 2)), '1b77354d-d021-410f-978f-d5a60d389819']

During handling of the above exception, another exception occurred:

[0;31mFileNotFoundError[0m                         Traceback (most recent call last)
Cell [0;32mIn[2], line 1[0m
[0;32m----> 1[0m be [38;5;241m=[39m [43mxr[49m[38;5;241;43m.[39;49m[43mopen_dataset[49m[43m([49m[38;5;124;43m'[39;49m[38;5;124;43m../../Land_and_Ocean_LatLong1.nc[39;49m[38;5;124;43m'[39;49m[43m)[49m
[1;32m      2[0m [38;5;66;03m#be = xr.open_dataset('/Users/vonw/work/courses/2021-2022/spring/CE401/jupyterbook/CE401/Land_and_Ocean_LatLong1.nc')[39;00m
[1;32m      3[0m be

File [0;32m/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/xarray/backends/api.py:541[0m, in [0;36mopen_dataset[0;34m(filename_or_obj, engine, chunks, cache, decode_cf, mask_and_scale, decode_times, decode_timedelta, use_cftime, concat_characters, decode_coords, drop_variables, inline_array, backend_kwargs, **kwargs)[0m
[1;32m    529[0m decoders [38;5;241m=[39m _resolve_decoders_kwargs(
[1;32m    530[0m     decode_cf,
[1;32m    531[0m     open_backend_dataset_parameters[38;5;241m=[39mbackend[38;5;241m.[39mopen_dataset_parameters,
[0;32m   (...)[0m
[1;32m    537[0m     decode_coords[38;5;241m=[39mdecode_coords,
[1;32m    538[0m )
[1;32m    540[0m overwrite_encoded_chunks [38;5;241m=[39m kwargs[38;5;241m.[39mpop([38;5;124m"[39m[38;5;124moverwrite_encoded_chunks[39m[38;5;124m"[39m, [38;5;28;01mNone[39;00m)
[0;32m--> 541[0m backend_ds [38;5;241m=[39m [43mbackend[49m[38;5;241;43m.[39;49m[43mopen_dataset[49m[43m([49m
[1;32m    542[0m [43m    [49m[43mfilename_or_obj[49m[43m,[49m
[1;32m    543[0m [43m    [49m[43mdrop_variables[49m[38;5;241;43m=[39;49m[43mdrop_variables[49m[43m,[49m
[1;32m    544[0m [43m    [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mdecoders[49m[43m,[49m
[1;32m    545[0m [43m    [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m,[49m
[1;32m    546[0m [43m[49m[43m)[49m
[1;32m    547[0m ds [38;5;241m=[39m _dataset_from_backend_dataset(
[1;32m    548[0m     backend_ds,
[1;32m    549[0m     filename_or_obj,
[0;32m   (...)[0m
[1;32m    557[0m     [38;5;241m*[39m[38;5;241m*[39mkwargs,
[1;32m    558[0m )
[1;32m    559[0m [38;5;28;01mreturn[39;00m ds

File [0;32m/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/xarray/backends/scipy_.py:307[0m, in [0;36mScipyBackendEntrypoint.open_dataset[0;34m(self, filename_or_obj, mask_and_scale, decode_times, concat_characters, decode_coords, drop_variables, use_cftime, decode_timedelta, mode, format, group, mmap, lock)[0m
[1;32m    305[0m store_entrypoint [38;5;241m=[39m StoreBackendEntrypoint()
[1;32m    306[0m [38;5;28;01mwith[39;00m close_on_error(store):
[0;32m--> 307[0m     ds [38;5;241m=[39m [43mstore_entrypoint[49m[38;5;241;43m.[39;49m[43mopen_dataset[49m[43m([49m
[1;32m    308[0m [43m        [49m[43mstore[49m[43m,[49m
[1;32m    309[0m [43m        [49m[43mmask_and_scale[49m[38;5;241;43m=[39;49m[43mmask_and_scale[49m[43m,[49m
[1;32m    310[0m [43m        [49m[43mdecode_times[49m[38;5;241;43m=[39;49m[43mdecode_times[49m[43m,[49m
[1;32m    311[0m [43m        [49m[43mconcat_characters[49m[38;5;241;43m=[39;49m[43mconcat_characters[49m[43m,[49m
[1;32m    312[0m [43m        [49m[43mdecode_coords[49m[38;5;241;43m=[39;49m[43mdecode_coords[49m[43m,[49m
[1;32m    313[0m [43m        [49m[43mdrop_variables[49m[38;5;241;43m=[39;49m[43mdrop_variables[49m[43m,[49m
[1;32m    314[0m [43m        [49m[43muse_cftime[49m[38;5;241;43m=[39;49m[43muse_cftime[49m[43m,[49m
[1;32m    315[0m [43m        [49m[43mdecode_timedelta[49m[38;5;241;43m=[39;49m[43mdecode_timedelta[49m[43m,[49m
[1;32m    316[0m [43m    [49m[43m)[49m
[1;32m    317[0m [38;5;28;01mreturn[39;00m ds

File [0;32m/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/xarray/backends/store.py:32[0m, in [0;36mStoreBackendEntrypoint.open_dataset[0;34m(self, store, mask_and_scale, decode_times, concat_characters, decode_coords, drop_variables, use_cftime, decode_timedelta)[0m
[1;32m     20[0m [38;5;28;01mdef[39;00m [38;5;21mopen_dataset[39m(
[1;32m     21[0m     [38;5;28mself[39m,
[1;32m     22[0m     store,
[0;32m   (...)[0m
[1;32m     30[0m     decode_timedelta[38;5;241m=[39m[38;5;28;01mNone[39;00m,
[1;32m     31[0m ):
[0;32m---> 32[0m     [38;5;28mvars[39m, attrs [38;5;241m=[39m [43mstore[49m[38;5;241;43m.[39;49m[43mload[49m[43m([49m[43m)[49m
[1;32m     33[0m     encoding [38;5;241m=[39m store[38;5;241m.[39mget_encoding()
[1;32m     35[0m     [38;5;28mvars[39m, attrs, coord_names [38;5;241m=[39m conventions[38;5;241m.[39mdecode_cf_variables(
[1;32m     36[0m         [38;5;28mvars[39m,
[1;32m     37[0m         attrs,
[0;32m   (...)[0m
[1;32m     44[0m         decode_timedelta[38;5;241m=[39mdecode_timedelta,
[1;32m     45[0m     )

File [0;32m/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/xarray/backends/common.py:128[0m, in [0;36mAbstractDataStore.load[0;34m(self)[0m
[1;32m    106[0m [38;5;28;01mdef[39;00m [38;5;21mload[39m([38;5;28mself[39m):
[1;32m    107[0m [38;5;250m    [39m[38;5;124;03m"""[39;00m
[1;32m    108[0m [38;5;124;03m    This loads the variables and attributes simultaneously.[39;00m
[1;32m    109[0m [38;5;124;03m    A centralized loading function makes it easier to create[39;00m
[0;32m   (...)[0m
[1;32m    125[0m [38;5;124;03m    are requested, so care should be taken to make sure its fast.[39;00m
[1;32m    126[0m [38;5;124;03m    """[39;00m
[1;32m    127[0m     variables [38;5;241m=[39m FrozenDict(
[0;32m--> 128[0m         (_decode_variable_name(k), v) [38;5;28;01mfor[39;00m k, v [38;5;129;01min[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mget_variables[49m[43m([49m[43m)[49m[38;5;241m.[39mitems()
[1;32m    129[0m     )
[1;32m    130[0m     attributes [38;5;241m=[39m FrozenDict([38;5;28mself[39m[38;5;241m.[39mget_attrs())
[1;32m    131[0m     [38;5;28;01mreturn[39;00m variables, attributes

File [0;32m/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/xarray/backends/scipy_.py:174[0m, in [0;36mScipyDataStore.get_variables[0;34m(self)[0m
[1;32m    172[0m [38;5;28;01mdef[39;00m [38;5;21mget_variables[39m([38;5;28mself[39m):
[1;32m    173[0m     [38;5;28;01mreturn[39;00m FrozenDict(
[0;32m--> 174[0m         (k, [38;5;28mself[39m[38;5;241m.[39mopen_store_variable(k, v)) [38;5;28;01mfor[39;00m k, v [38;5;129;01min[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mds[49m[38;5;241m.[39mvariables[38;5;241m.[39mitems()
[1;32m    175[0m     )

File [0;32m/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/xarray/backends/scipy_.py:163[0m, in [0;36mScipyDataStore.ds[0;34m(self)[0m
[1;32m    161[0m [38;5;129m@property[39m
[1;32m    162[0m [38;5;28;01mdef[39;00m [38;5;21mds[39m([38;5;28mself[39m):
[0;32m--> 163[0m     [38;5;28;01mreturn[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_manager[49m[38;5;241;43m.[39;49m[43macquire[49m[43m([49m[43m)[49m

File [0;32m/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/xarray/backends/file_manager.py:191[0m, in [0;36mCachingFileManager.acquire[0;34m(self, needs_lock)[0m
[1;32m    176[0m [38;5;28;01mdef[39;00m [38;5;21macquire[39m([38;5;28mself[39m, needs_lock[38;5;241m=[39m[38;5;28;01mTrue[39;00m):
[1;32m    177[0m [38;5;250m    [39m[38;5;124;03m"""Acquire a file object from the manager.[39;00m
[1;32m    178[0m 
[1;32m    179[0m [38;5;124;03m    A new file is only opened if it has expired from the[39;00m
[0;32m   (...)[0m
[1;32m    189[0m [38;5;124;03m        An open file object, as returned by ``opener(*args, **kwargs)``.[39;00m
[1;32m    190[0m [38;5;124;03m    """[39;00m
[0;32m--> 191[0m     file, _ [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_acquire_with_cache_info[49m[43m([49m[43mneeds_lock[49m[43m)[49m
[1;32m    192[0m     [38;5;28;01mreturn[39;00m file

File [0;32m/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/xarray/backends/file_manager.py:215[0m, in [0;36mCachingFileManager._acquire_with_cache_info[0;34m(self, needs_lock)[0m
[1;32m    213[0m     kwargs [38;5;241m=[39m kwargs[38;5;241m.[39mcopy()
[1;32m    214[0m     kwargs[[38;5;124m"[39m[38;5;124mmode[39m[38;5;124m"[39m] [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39m_mode
[0;32m--> 215[0m file [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_opener[49m[43m([49m[38;5;241;43m*[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_args[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m
[1;32m    216[0m [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39m_mode [38;5;241m==[39m [38;5;124m"[39m[38;5;124mw[39m[38;5;124m"[39m:
[1;32m    217[0m     [38;5;66;03m# ensure file doesn't get overridden when opened again[39;00m
[1;32m    218[0m     [38;5;28mself[39m[38;5;241m.[39m_mode [38;5;241m=[39m [38;5;124m"[39m[38;5;124ma[39m[38;5;124m"[39m

File [0;32m/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/xarray/backends/scipy_.py:102[0m, in [0;36m_open_scipy_netcdf[0;34m(filename, mode, mmap, version)[0m
[1;32m     99[0m     filename [38;5;241m=[39m io[38;5;241m.[39mBytesIO(filename)
[1;32m    101[0m [38;5;28;01mtry[39;00m:
[0;32m--> 102[0m     [38;5;28;01mreturn[39;00m [43mscipy[49m[38;5;241;43m.[39;49m[43mio[49m[38;5;241;43m.[39;49m[43mnetcdf_file[49m[43m([49m[43mfilename[49m[43m,[49m[43m [49m[43mmode[49m[38;5;241;43m=[39;49m[43mmode[49m[43m,[49m[43m [49m[43mmmap[49m[38;5;241;43m=[39;49m[43mmmap[49m[43m,[49m[43m [49m[43mversion[49m[38;5;241;43m=[39;49m[43mversion[49m[43m)[49m
[1;32m    103[0m [38;5;28;01mexcept[39;00m [38;5;167;01mTypeError[39;00m [38;5;28;01mas[39;00m e:  [38;5;66;03m# netcdf3 message is obscure in this case[39;00m
[1;32m    104[0m     errmsg [38;5;241m=[39m e[38;5;241m.[39margs[[38;5;241m0[39m]

File [0;32m/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/scipy/io/_netcdf.py:246[0m, in [0;36mnetcdf_file.__init__[0;34m(self, filename, mode, mmap, version, maskandscale)[0m
[1;32m    244[0m [38;5;28mself[39m[38;5;241m.[39mfilename [38;5;241m=[39m filename
[1;32m    245[0m omode [38;5;241m=[39m [38;5;124m'[39m[38;5;124mr+[39m[38;5;124m'[39m [38;5;28;01mif[39;00m mode [38;5;241m==[39m [38;5;124m'[39m[38;5;124ma[39m[38;5;124m'[39m [38;5;28;01melse[39;00m mode
[0;32m--> 246[0m [38;5;28mself[39m[38;5;241m.[39mfp [38;5;241m=[39m [38;5;28;43mopen[39;49m[43m([49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mfilename[49m[43m,[49m[43m [49m[38;5;124;43m'[39;49m[38;5;132;43;01m%s[39;49;00m[38;5;124;43mb[39;49m[38;5;124;43m'[39;49m[43m [49m[38;5;241;43m%[39;49m[43m [49m[43momode[49m[43m)[49m
[1;32m    247[0m [38;5;28;01mif[39;00m mmap [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:
[1;32m    248[0m     [38;5;66;03m# Mmapped files on PyPy cannot be usually closed[39;00m
[1;32m    249[0m     [38;5;66;03m# before the GC runs, so it's better to use mmap=False[39;00m
[1;32m    250[0m     [38;5;66;03m# as the default.[39;00m
[1;32m    251[0m     mmap [38;5;241m=[39m ([38;5;129;01mnot[39;00m IS_PYPY)

[0;31mFileNotFoundError[0m: [Errno 2] No such file or directory: '/home/runner/work/CE401/Land_and_Ocean_LatLong1.nc'

